src/jdk.compiler/share/classes/com/sun/tools/javac/comp/Annotate.java

```diff
@@ -1093,32 +1075,30 @@ public void annotateTypeParameterSecondStage(JCTree tree, List<JCAnnotation> ann
     private class TypeAnnotate extends TreeScanner {
         private final Env<AttrContext> env;
         private final Symbol sym;
-        private JCTree deferDecl;
 
-        public TypeAnnotate(Env<AttrContext> env, Symbol sym, JCTree deferDecl) {
+        public TypeAnnotate(Env<AttrContext> env, Symbol sym) {
```

Comment:
All these changes are truly great, as they are sign that you were able to remove a lot of coupling from the way things were implemented. E.g. the goal of this PR is not to "make annotation processing simpler", but turns out that, by centralizing the lint mapping logic in one place, other parts of the compiler also got simpler!